src/analytics.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Analytics for server-side events.
+ *
+ * @module analytics
+ */
+import { default as UserAgentBag } from "user-agent-bag";
+import { PLAUSIBLE_ANALYTICS_DOMAIN, PLAUSIBLE_ANALYTICS_ENDPOINT } from "./config.js";
+import { fetch } from "./fetch.js";
+
+
+/**
+ * Generate middleware to record an analytics event via a request to
+ * Plausible's API.
+ *
+ * Intended for use in an Express routing chain.  The generated middleware
+ * function does not block on the analytics request before returning.
+ *
+ * If the User-Agent of the incoming request starts with "Nextstrain-CLI/",
+ * then it is parsed for several bits of information and those are
+ * automatically attached to the analytics event as custom properties.
+ *
+ * @function recordEvent
+ * @param {object} data
+ * @param {string} data.name - Event name.  Defaults to "pageview" (Plausible's default).
+ * @param {object} data.props - Custom properties.  Defaults to {}.
+ * @param {boolean} data.interactive - Interactive session or not?  Defaults to
+ *   false (opposite of Plausible's default), with the reasoning that most events
+ *   we're likely to record server-side aren't for interactive browser-based
+ *   sessions.
+ * @returns {expressMiddlewareAsync}
+ */
+export const recordEvent = ({name = "pageview", props = {}, interactive = false} = {}) => async (req, res, next) => {
+  if (PLAUSIBLE_ANALYTICS_DOMAIN) {
+    /* We intentionally do not "await fetch()" as we do not want to block
+     * request processing on analytics; we'd rather ignore failures.
+     *    -trs, 28 April 2025
+     */
+    fetch(
```

Comment:
I'm a little hesitant to use the interpreter's event loop/task queue here as our own offline work queue, as it seems like badness awaits if we start clogging it up with requests that take a long time to complete (e.g. if network cxn to Plausible is very slow to connect or transfer data). And it's not a queue we can manage on our own or have much insight into by default. But it seems an ok place to start, probably, esp. in the absence of some other obvious work queue in our stack.